x-pack/plugins/ml/public/application/data_frame_analytics/common/fields.ts

```diff
@@ -260,16 +261,22 @@ export const getDefaultFieldsFromJobCaps = (
   const predictedField = `${resultsField}.${
     predictionFieldName ? predictionFieldName : defaultPredictionField
   }`;
-
-  const allFields: any = [
-    {
-      id: `${resultsField}.is_training`,
-      name: `${resultsField}.is_training`,
-      type: ES_FIELD_TYPES.BOOLEAN,
-    },
-    { id: predictedField, name: predictedField, type },
-    ...fields,
-  ].sort(({ name: a }, { name: b }) => sortRegressionResultsFields(a, b, jobConfig));
+  // Only need to add these first two fields if we didn't use dest index pattern to get the fields
+  const allFields: any =
+    needsDestIndexFields === true
+      ? [
+          {
+            id: `${resultsField}.is_training`,
+            name: `${resultsField}.is_training`,
+            type: ES_FIELD_TYPES.BOOLEAN,
+          },
+          { id: predictedField, name: predictedField, type },
+        ]
+      : [];
+
+  allFields.push(...fields);
+  // @ts-ignore
```

Comment:
it would be good to have a comment explaining what is being ignored on this line.